frontend/react/src/infrastructure/components/QueryBoundary/query/useQueryBoundaryFn.test.tsx

```diff
@@ -0,0 +1,127 @@
+import { renderHook } from '@testing-library/react';
+import {
+  type QueryBoundaryFnProps,
+  useQueryBoundaryFn,
+} from './useQueryBoundaryFn.ts';
+import { useQuery } from '@tanstack/react-query';
+
+jest.mock('@tanstack/react-query', () => ({
+  useQuery: jest.fn(),
+}));
+
+describe('useQueryBoundary', () => {
+  const defaultTanstackQueryResult = {
+    isError: false,
+    isFetching: false,
+    isSuccess: false,
+    data: '',
+  };
+  const defaultQueryFnProps: QueryBoundaryFnProps<string> = {
+    defaultQueryDataResult: '',
+    queryFn: jest.fn(),
+    queryKey: ['myQueryKey'],
+  };
+
+  const tanstackUseQuery = useQuery as jest.MockedFunction<typeof useQuery>;
+
+  beforeEach(() => jest.clearAllMocks());
+
+  it('Should return fetching', () => {
+    tanstackUseQuery.mockReturnValue({
+      ...defaultTanstackQueryResult,
+      isFetching: true,
+    } as never);
```

Comment:
Joa der Type passt hier nicht wirklich. Es wird schon genutzt. Zudem sollte `satisfies` doch hier auch funktionieren und dann verlierst du die Typesicherheit nicht.

Du könntest auch `tanstackUseQuery` weiter typisieren, sodass der Rückgabetype dort schon bekannt ist.
Vermutlich so in der Art: 
` const tanstackUseQuery = useQuery as jest.MockedFunction<typeof useQuery<QueryBoundaryFnResult, Error>>`